pkg/issuer/acme/dns/dns.go

```diff
@@ -91,39 +91,29 @@ func (s *Solver) Present(ctx context.Context, issuer v1alpha1.GenericIssuer, ch
 }
 
 // Check verifies that the DNS records for the ACME challenge have propagated.
-func (s *Solver) Check(ctx context.Context, issuer v1alpha1.GenericIssuer, ch *v1alpha1.Challenge) (bool, error) {
-	providerName := ch.Spec.Config.DNS01.Provider
-	if providerName == "" {
-		return false, fmt.Errorf("dns01 challenge provider name must be set")
-	}
+func (s *Solver) Check(ctx context.Context, issuer v1alpha1.GenericIssuer, ch *v1alpha1.Challenge) error {
 
-	providerConfig, err := issuer.GetSpec().ACME.DNS01.Provider(providerName)
+	fqdn, value, ttl, err := util.DNS01Record(ch.Spec.DNSName, ch.Spec.Key, s.DNS01Nameservers, false)
 	if err != nil {
-		return false, err
-	}
-
-	fqdn, value, ttl, err := util.DNS01Record(ch.Spec.DNSName, ch.Spec.Key, s.DNS01Nameservers, followCNAME(providerConfig.CNAMEStrategy))
-	if err != nil {
-		return false, err
+		return err
 	}
 
 	glog.Infof("Checking DNS propagation for %q using name servers: %v", ch.Spec.DNSName, s.Context.DNS01Nameservers)
 
 	ok, err := util.PreCheckDNS(fqdn, value, s.Context.DNS01Nameservers,
```

Comment:
This call here is what performs the actual self check - I *think* we need to make a similar change to `PreCheckDNS` as we have done to this function, to make it only return an error type. I may be wrong though?!

Otherwise the only information we'll ever surface is "DNS record for 'example.com' not yet propagated" (instead of some helpful error surfaced from the actual checker)

Reply:
PreCheckDNS turns various conditions into errors, so the only time we'll get the "not yet propagated" error is when we had no errors but the TXT record was missing